nav2_collision_monitor/src/polygon.cpp

```diff
@@ -392,6 +433,30 @@ bool Polygon::getCommonParameters(
           node->get_parameter(polygon_name_ + ".footprint_topic").as_string();
       }
     }
+
+    // By default, use all observation sources for polygon
+    const std::vector<std::string> default_observation_sources;
+    nav2_util::declare_parameter_if_not_declared(
+      node, "observation_sources", rclcpp::ParameterValue(default_observation_sources));
```

Comment:
By the time this is called, shouldn't we know that observiation_sources is already declared? We could be able to get just `get_parameter` it, no?

Reply:
Yes, the parameter `observation_sources` is already declared when node is getting polygons' parameters.

The problem is the tests. In `polygons_test.cpp`, we don't use any collision monitor node but only the polygons. So not declaring parameter `observation_sources` here with a default value results in many tests failures. 

If we consider that the polygon as an independent structure, so that its methods `configure()` , `getParameters()` and `getCommonParameter()` can be successfully called without needing any collision monitor node, then we need to keep this line. Since the polygons are tested independently, I though this was the more reasonable.

But if we don't want to risk any double declaration of param `observation_sources`, we can remove this line from `polygon.cpp` and add it to  `polygons_test.cpp` so that the parameter will be declared first, only for these tests

Reply:
I don't mind its re-declaration, however the original is a required input (i.e. only specifies type, no default value, it must be specified) 

```
    nav2_util::declare_parameter_if_not_declared(
      node, "observation_sources", rclcpp::PARAMETER_STRING_ARRAY);
```

Here, you say the default value is none, which is not the same expression. We'd need them to be the same if we're going to re-declare in case this module is used independently so that they act the same. If you do that, then you'd still need to declare or set the value of `observation_sources` in the tests before running the polygon work so that it could `get_parameter` something without throwing the unset-parameter exception. 

So, I think you can declare again, but it should be  `PARAMETER_STRING_ARRAY`, and then the tests need to still declare or set it to some value

Reply:
>  declare again, but it should be PARAMETER_STRING_ARRAY, and then the tests need to still declare or set it to some value

It's done